backend/src/core/routes/settings.routes.ts

```diff
@@ -27,6 +27,12 @@ const demoDisplayedValidator = {
   }),
 }
 
+const updateAnnouncementVersionValidator = {
+  [Segments.BODY]: Joi.object({
+    announcement_version: Joi.string().default(null),
```

Comment:
This needs to be validated. You may want to make it required. In addition, you should also check that the value 
- Has the right format (X.Y.Z)
- Is less than or equals to the current version

Otherwise a user might be able to set this to an invalid value through the API.

Reply:
Did some simple regex for the first, but for the second check - Is less than or equals to the current version - there would need to either be a DB call to check the current version (which I believe Joi can't do), or I can change the endpoint to receive two versions instead (current, lastSeen). But I'm not sure if that would necessarily provide the additional protection we want?

Reply:
I think should be able to retrieve the current package version using `process.env.npm_package_version`. https://stackoverflow.com/a/48619640. Mainly want to avoid being able to set the user's version to something greater than the current package version. Otherwise, they will miss all future announcements.